### PR TITLE
Log in

### PR DESCRIPTION
**Pull request by hazwanizafirah**

### CODE REVIEW

**Comment by hazwanizafirah** [comment]
Ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,27 @@
 class User < ApplicationRecord
+  has_secure_password
```
Comment (huongnt-2545) [inline app/models/user.rb:2]: move this line place to after the validate area

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   before_save {self.email = email.downcase}
```
Comment (huongnt-2545) [inline app/models/user.rb:5]: remove this file, you already define at line 10

File: app/views/sessions/create.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#create</h1>
```
Comment (huongnt-2545) [inline app/views/sessions/create.html.erb:1]: delete this file if it's not useful

File: app/views/sessions/destroy.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#destroy</h1>
```
Comment (huongnt-2545) [inline app/views/sessions/destroy.html.erb:1]: delete this file if it's not useful

**Comment by hazwanizafirah** [comment]
Ready

**Comment by hazwanizafirah** [comment]
ready

File: app/models/user.rb [hunk 5]
```diff
@@ -1,9 +1,26 @@
 class User < ApplicationRecord
+
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   before_save {self.email = email.downcase}
   validates :name, presence: true
   validates :email, presence: true, length: {maximum: 255},
   format: {with: VALID_EMAIL_REGEX}
 
-  has_secure_password
```
Comment (huongnt-2545) [inline app/models/user.rb:8]: why do you remove this line?

**Review by huongnt-2545 — COMMENTED** [review]
This pull still has comment not fix yet

File: app/models/user.rb (hunk 2 above)
Comment (hazwanizafirah) [reply app/models/user.rb:4]: sorry but which line should I delete? is it has_secure_password?

**Comment by hazwanizafirah** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**